File: python/stock_picker.py
def picker(prices):

    # Kadane's works here, we initialize the max profit on -inf
    
    max_profit = float("-inf")
    curr = 0
    coordinates = []

    # we get all coordinates where j is after i to avoid the sell before buy

    for i in range(len(prices)):
        for j in range(i,len(prices)):
            profit = prices[j] - prices[i]
            # if the current profit is better than max, we update the max and replace the coordinates
            curr = max (profit,curr)
            if curr > max_profit:
                max_profit = max(max_profit,curr)
                coordinates = [i,j]
    return coordinates

File: js/stockPicker.js

exports.picker = function(prices) {

    let max_profit = -Infinity
    let curr = 0
    let coordinates = []

    for (let i = 0; i < prices.length; i++){
        for (let j = i; j < prices.length; j++){
            profit = prices[j] - prices[i]
            curr = Math.max(profit,curr)
            if (curr > max_profit){
                max_profit = Math.max(max_profit,curr)
                coordinates = [i,j]
            }
        }
    }
    return coordinates
}
